cmd/config/internal/commands/cmdset.go

```diff
@@ -126,15 +130,53 @@ func (r *SetRunner) preRunE(c *cobra.Command, args []string) error {
 			return err
 		}
 	}
-
 	return nil
 }
 
 func (r *SetRunner) runE(c *cobra.Command, args []string) error {
 	if setterVersion == "v2" {
-		count, err := r.Set.Set(r.OpenAPIFile, args[0])
-		fmt.Fprintf(c.OutOrStdout(), "set %d fields\n", count)
-		return handleError(c, err)
+		openAPIFileName, err := ext.OpenAPIFileName()
+		if err != nil {
+			return err
+		}
+
+		r.Set.OpenAPIFileName = openAPIFileName
+		resourcePackagesPaths, err := pathutil.DirsWithFile(args[0], openAPIFileName, r.Set.RecurseSubPackages)
+		if err != nil {
+			return err
+		}
+
+		if len(resourcePackagesPaths) == 0 {
+			return errors.Errorf("unable to find %s in package %s", r.Set.OpenAPIFileName, args[0])
+		}
+
+		for _, resourcesPath := range resourcePackagesPaths {
+			r.Set = settersutil.FieldSetter{
+				Name:               r.Set.Name,
+				Value:              r.Set.Value,
+				ListValues:         r.Set.ListValues,
+				Description:        r.Set.Description,
+				SetBy:              r.Set.SetBy,
+				Count:              0,
+				OpenAPIPath:        filepath.Join(resourcesPath, openAPIFileName),
+				OpenAPIFileName:    openAPIFileName,
+				ResourcesPath:      resourcesPath,
+				RecurseSubPackages: r.Set.RecurseSubPackages,
```

Comment:
Since we are creating a new FieldSetter for each package/subpackage, why would we need to pass this along? It seems like a FieldSetter only needs to handle one package.

Reply:
So that the behavior is consistent both for cli and api. It is not being used now at the api but wanted to be consistent.

Reply:
Yeah, I'm ok with leaving this for now. But we should clean this up in a follow-up PR.